baal/active/dataset/base.py

```diff
@@ -0,0 +1,126 @@
+import warnings
+from typing import Union, List, Optional, Any, Iterable, Sized
+
+import numpy as np
+from sklearn.utils import check_random_state
+from torch.utils import data as torchdata
+
+
+class SplittedDataset(torchdata.Dataset):
+    """Abstract class for Dataset that can be splitted.
+
+    Args:
+        labelled: An array that acts as a mask which is greater than 1 for every
+            data point that is labelled, and 0 for every data point that is not
+            labelled.
+        random_state: Set the random seed for label_randomly().
+        last_active_steps: If specified, will iterate over the last active steps
+                            instead of the full dataset. Useful when doing partial finetuning.
+    """
+
+    def __init__(
+        self,
+        labelled,
+        random_state=None,
+        last_active_steps: int = -1,
+    ) -> None:
+        self.labelled_map = labelled
+        self.random_state = check_random_state(random_state)
+        if last_active_steps == 0 or last_active_steps < -1:
+            raise ValueError("last_active_steps must be > 0 or -1 when disabled.")
+        self.last_active_steps = last_active_steps
+
+    def get_indices_for_active_step(self):
+        """Returns the indices required for the active step.
+
+        Returns the indices of the labelled items. Also takes into account self.last_active_step.
+
+        Returns:
+            Array of the selected indices for training.
+        """
+        if self.last_active_steps == -1:
+            min_labelled_step = 0
+        else:
+            min_labelled_step = max(0, self.current_al_step - self.last_active_steps)
+        indices = np.arange(len(self.labelled_map))
+        bool_mask = self.labelled_map > min_labelled_step
```

Comment:
is the purpose of this function to check what are the indices to be labelled until the end of AL process or which indices should be labelled for this specific step? it seems to me that it provides the first output but the documentation kinda says that we do the second on ? also in any case, what is the use case?

Reply:
```suggestion
        bool_mask = self.labelled_map > min_labelled_step
```
I am confused by this specifically. `min_labelled_step` can be any integer. that means that we only get the data points that are labelled after a certain active_step ? and when we call this function in `len` don't we want all the available labels?

Reply:
This method specifies which indices should be selected for training at this step.

when we call len() we want the size of the Dataset I think? So in this case it would be all items selected for training. If we want the number of labelled items, we should call ActiveLearningDataset.n_labelled in my opinion.

Reply:
ok im clear now great 👍